realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -211,6 +211,35 @@ public void testMoveUp_nonManagedMode() {
         assertEquals(oldIndex, owner.getDogs().indexOf(dog));
     }
 
+    public void testFirst_nonManagedMode() {
+        RealmList<AllTypes> list = new RealmList<AllTypes>();
+        AllTypes object1 = new AllTypes();
+        AllTypes object2 = new AllTypes();
+        list.add(object1);
+        list.add(object2);
+
+        assertEquals(object1, list.first());
+    }
+
+    public void testFirstEmptyList_nonManagedMode() {
```

Comment:
I would merge this with the other testing last. So having the empty list, verify that all methods behaves correctly (also size()).